File: Convoy Caravan Code parts/Utils/LanguageManager.m
//
//  LanguageManager.m
//  Convoy Caravan
//
//  Created by Priba on 9/4/18.
//  Copyright © 2018 Priba. All rights reserved.
//

#import "LanguageManager.h"

@implementation LanguageManager

+ (instancetype)sharedInstance {
    static id instance;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        instance = [[self alloc] init];
    });
    return instance;
}

+ (NSString*)setLanguageString:(NSString*)string
{
    if([[NSUserDefaults standardUserDefaults] objectForKey:@"language_str"] != nil){
        NSString *str;
        NSString *path;
        
        path = [[NSBundle mainBundle] pathForResource:[self getLanguageExtension] ofType:@"lproj"];
        
        NSBundle * languageBundle = [NSBundle bundleWithPath:path];
        str = [languageBundle localizedStringForKey:string value:@"" table:nil];
        return str;
    }else{
        return NSLocalizedString(string, nil);
    }
    
}

+ (NSString*)getLanguageExtension
{
    if([[NSUserDefaults standardUserDefaults] objectForKey:@"language_str"] != nil){
        
        NSString *langStr = [[NSUserDefaults standardUserDefaults] objectForKey:@"language_str"];
        
        if([langStr isEqualToString:@"SW"]){
            return @"sw";
        }else{
            return @"en";
        }
    }else{
        NSString *language = [[NSLocale preferredLanguages] objectAtIndex:0];
        NSDictionary *languageDic = [NSLocale componentsFromLocaleIdentifier:language];
        NSString *languageCode = [languageDic objectForKey:@"kCFLocaleLanguageCodeKey"];
        
        return languageCode;
    }
}

@end

File: Linguado Code parts/Localization/LanguageManager.swift
//
//  LanguageManager.swift
//  JobDeal
//
//  Created by Priba on 12/7/18.
//  Copyright © 2018 Priba. All rights reserved.
//

import Foundation

class LanguageManager{
    
    static let sharedInstance = LanguageManager()
    
    public func getStringForKey(key:String, uppercased:Bool = false)-> String{
        
        var path = Bundle.main.path(forResource: getLanguageExtension(), ofType: "lproj")
        
        if path == nil {
            path = Bundle.main.path(forResource: "en", ofType: "lproj")
        }
        
        let langBundle = Bundle.init(path: path!)
        let str = langBundle?.localizedString(forKey: key, value: "", table: nil)
        
        if(uppercased){
            return str?.uppercased() ?? ""
        }
        
        return str ?? ""
        
    }
    
    public func getLanguageExtension()-> String{
            
        if let langStr = UserDefaults.standard.object(forKey: "language_str") as? String{
            return langStr.lowercased()
            
        }else{
            let language = Locale.preferredLanguages.first
            let languageDict = Locale.components(fromIdentifier: language ?? "")
            var languageCode = languageDict["kCFLocaleLanguageCodeKey"]
            
            if languageCode == "pt" { //REFACTOR
                languageCode = "pt-PT"
            }
            
            return languageCode ?? ""
        }
        
    }
    
    
}
